addons/mass_mailing_sms/models/sms_sms.py

```diff
@@ -22,9 +22,9 @@ def _update_body_short_links(self):
                 continue
 
             body = sms.body
-            for url in re.findall(tools.TEXT_URL_REGEX, body):
+            for url in set(re.findall(tools.TEXT_URL_REGEX, body)):
                 if url.startswith(sms.get_base_url() + '/r/'):
-                    body = body.replace(url, url + '/s/%s' % sms.id)
+                    body = re.sub(url + r'(?![\w@:%.+&~#=/-])', url + f'/s/{sms.id}', body)
```

Comment:
shouldn't `url` be regex-escaped here, when constructing the pattern for `res.sub`?

Reply:
Indeed, it should, thanks!